driver/src/test/java/org/neo4j/driver/integration/SSLSocketChannelIT.java

```diff
@@ -92,22 +111,73 @@ public void shouldPerformTLSHandshake() throws Throwable
     }
 
     @Test
-    public void shouldRejectServerConnectionDueToWrongCert() throws Throwable
+    public void shouldFailTLSHandshakeDueToWrongCertInKnownCertsFile() throws Throwable
+    {
+        // Given
+        SocketChannel channel = SocketChannel.open();
+        channel.connect( new InetSocketAddress( "localhost", 7687 ) );
+        File knownCerts = File.createTempFile( "neo4j_known_certs", ".tmp" );
+        knownCerts.deleteOnExit();
+
+        //create a Fake Cert for the server in knownCert
+        createFakeServerCertPairInKnownCerts( "localhost", 7687, knownCerts );
+
+        // When & Then
+        SSLSocketChannel sslChannel = null;
+        try
+        {
+            sslChannel = new SSLSocketChannel( "localhost", 7687, channel, mock( Logger.class ),
+                    usingKnownCerts( knownCerts ) );
+            sslChannel.close();
+        }
+        catch ( SSLHandshakeException e )
+        {
+            assertEquals( "General SSLEngine problem", e.getMessage() );
+            assertEquals( "General SSLEngine problem", e.getCause().getMessage() );
+            assertTrue( e.getCause().getCause().getMessage().startsWith(
+                    "The certificate received from the server is different from the one we've known in file" ) );
```

Comment:
From the perspective of a user that is not well-versed with TLS or trust-on-first-use, it would be awesome to make tailor this message to helping the user with how to fix the problem. Eg, could we change this message to be something more along the lines of:

    Unable to connect to neo4j at `neo4j://blahblah`, because the encryption certificate the server uses has changed. This is a security feature to protect against man-in-the-middle attacks. 

    If you trust the certificate the server uses now, simply remove the line that starts with <blah> in the file <bleh>. The certificate used by the server is: <cert info>.

I know that's a really long exception message, but I don't know that that's really an issue - independent of which, main point being that we give the user a hand in explaining how to resolve it.